lightning/src/ln/msgs.rs

```diff
@@ -1787,6 +1789,29 @@ impl fmt::Debug for OnionPacket {
 	}
 }
 
+/// BOLT 4 onion packet including hop data for the next peer.
+#[derive(Clone, Hash, PartialEq, Eq)]
+pub struct VariableLengthOnionPacket {
```

Comment:
As per [BOLT#9](https://github.com/lightning/bolts/blob/trampoline-onion/04-onion-routing.md#trampoline-payments), the Trampoline is correctly introduced as a `VariableLengthOnionPacket`

```
The trampoline_onion_packet has a variable size to allow...
```

The struct looks good and is correctly implemented. And, I think it can use a little nudge on the documentation side so that its purpose is explained in more detail and it can be easily distinguishable on a documentation level from struct `OnionPacket`.